### PR TITLE
refactor api.utils.model_view

### PR DESCRIPTION
**Pull request by xiaoxianma**

### CODE REVIEW

File: api/utils.py [hunk 1]
```diff
@@ -127,220 +128,299 @@ async def __call__(self, request: Request, security_scopes: SecurityScopes, retu
...
-            return await pagination.paginate(orm_model, user.id)
-
-    async def get_count(user: Union[None, schemes.User] = Security(auth_dependency, scopes=scopes["get_count"])):
-        return (
-            await (
-                (orm_model.query.where(orm_model.user_id == user.id) if orm_model != models.User else orm_model.query)
-                .with_only_columns([db.db.func.count(distinct(orm_model.id))])
-                .order_by(None)
-                .gino.scalar()
+    def _get(self):
```
Comment (xiaoxianma) [inline api/utils.py:270]: Not sure why the code below doesn't work:
```python
async def get(pagination: pagination.Pagination = Depends(), user: Union[None, schemes.User] = None):
    user = user if user else Security(self.auth_dependency, scopes=self.scopes["get_all"])
```

Comment (xiaoxianma) [reply api/utils.py:270]: That's why I have to put it into a wrapper function

File: api/utils.py [hunk 2]
```diff
@@ -127,220 +128,299 @@ async def __call__(self, request: Request, security_scopes: SecurityScopes, retu
...
-        user: Union[None, schemes.User] = Security(auth_dependency, scopes=scopes["put"]),
-    ):  # type: ignore
-        item = await _get_one(model_id, user, True)
-        try:
-            if custom_methods.get("put"):
-                await custom_methods["put"](item, model, user)  # pragma: no cover
+        return await self._get_one(model_id, user)
+
+    def _post(self):
+        model_type = self.create_model
```
Comment (xiaoxianma) [inline api/utils.py:312]: Have to create a function wrapper to make model_type lazy-loaded.

File: api/utils.py (hunk 1 above)
Comment (MrNaif2018) [reply api/utils.py:270]: You could use
```
try:
     user = await self.auth_dependency(request, SecurityScopes(self.scopes["get_all"]))
except HTTPException:
    user = None
```
But actually using a wrapper is cleaner

File: api/utils.py (hunk 2 above)
Comment (MrNaif2018) [reply api/utils.py:312]: From investigating, indeed, function wrapper seems to be the most clear solution.
But is creating `model_type` variable necessary here?

File: api/utils.py (hunk 1 above)
Comment (xiaoxianma) [reply api/utils.py:270]: I see. Thanks

**Review by MrNaif2018 — APPROVED** [review]
Thank you very much for your PR! Merging!

**Pull request merged**